src/main/java/org/jenkinsci/plugins/workflow/cps/actions/ArgumentsActionImpl.java

```diff
@@ -274,7 +273,7 @@ Object sanitizeObjectAndRecordMutation(@CheckForNull Object o, @CheckForNull Env
     @Nonnull
     Map<String,Object> sanitizeMapAndRecordMutation(@Nonnull Map<String, Object> mapContents, @CheckForNull EnvVars variables) {
         // Package scoped so we can test it directly
-        HashMap<String, Object> output = new HashMap<String, Object>();
+        HashMap<String, Object> output = Maps.newHashMapWithExpectedSize(mapContents.size());
```

Comment:
How is this related to the rest of the PR? Seems like something that should have been filed separately.

Reply:
@jglick It's due to the BulkFlowNodeStorage we now use as of this PR holding more nodes in memory (potentially).  This reduces the memory impact of that.